parachain/pallets/bridge/src/channel/inbound.rs

```diff
@@ -0,0 +1,116 @@
+use frame_support::{dispatch::DispatchResult, storage::StorageMap, traits::Get};
+use sp_core::H160;
+use sp_std::{cell::Cell, marker::PhantomData, boxed::Box};
+use artemis_core::{ChannelId, Message};
+use crate::{
+	Error,
+	Config,
+	InboundChannels,
+	primitives::{InboundChannel, InboundChannelData}
+};
+
+/// Construct an inbound channel object
+pub fn make_inbound_channel<T>(channel_id: ChannelId) -> Box<dyn InboundChannel<T::AccountId>>
+where
+	T: Config
+{
+	match channel_id {
+		ChannelId::Basic => Box::new(BasicInboundChannel::<T>::new()),
+		ChannelId::Incentivized => Box::new(IncentivizedInboundChannel::<T>::new()),
+	}
+}
+
+const DUMMY_APP_ID: H160 = H160::zero();
+
+/// Basic Channel
+struct BasicInboundChannel<T: Config> {
+	storage: Storage<T>
+}
+
+impl<T: Config> BasicInboundChannel<T> {
+	fn new() -> Self {
+		Self {
+			storage: Storage::new(ChannelId::Basic)
+		}
+	}
+}
+
+impl<T: Config> InboundChannel<T::AccountId> for BasicInboundChannel<T> {
+	fn submit(&mut self, relayer: &T::AccountId, message: &Message) -> DispatchResult {
+		self.storage.try_mutate(|data| {
+			// Example: Increment nonce
+			data.nonce += 1;
```

Comment:
need to check message.nonce?

Reply:
Added a comment to these `submit` methods indicating that they are still WIP. The goal of this PR is the overall design and API for channels in substrate. So the implementations of these methods will come eventually.